src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,477 @@
+        }
+
+        std::vector<Allocation::Ptr> free_allocations{};
+        for (const auto& alloc : allocations) {
+            if (!alloc->is_used &&
+                alloc_size <= alloc->size &&
+                (alloc->offset % requested_alignment) == 0) {
+                free_allocations.push_back(alloc);
+            }
+        }
+        std::sort(free_allocations.begin(), free_allocations.end(),
+            [](const Allocation::Ptr& alloc0, const Allocation::Ptr& alloc1) {
+                return alloc0->size < alloc1->size;
+            });
+        if (!free_allocations.empty()) {
+            const auto alloc = free_allocations.front();
+            alloc->is_used = true;
+            alloc->is_transaction = is_transaction;
+            return alloc;
+        } else {
+            size_t alloc_offset = 0;
+            if (requested_alignment > 1) {
+                alloc_offset = (requested_alignment - ((current_offset+alloc_size) % requested_alignment));
+            }
+
+            const size_t aligned_alloc_size = alloc_offset + alloc_size;
+            current_offset += aligned_alloc_size;
+            Xbyak::Address addr = code_generator.ptr[base_pointer - current_offset];
+            const auto alloc = std::make_shared<Allocation>(addr, current_offset, aligned_alloc_size);
+            alloc->is_transaction = is_transaction;
+            allocations.push_back(alloc);
+            return alloc;
+        }
+    }
+
+    void deallocate() {
+        while (!allocations.empty()) {
+            const auto& last = allocations.back();
+            if (last->is_used) {
+                break;
+            }
+            current_offset -= last->size;
+            allocations.pop_back();
+        }
+    }
+
+    void checkUnique(bool isCtor) {
+        static thread_local bool isCreated = false;
+        if (isCtor) {
+            if (isCreated) {
+                IE_THROW() << "There should be only one instance of StackAllocator per thread !!";
+            }
+            isCreated = true;
+        } else {
+            isCreated = false;
+        }
+    }
+
+    bool isTransaction() const {
+        return is_transaction_;
+    }
+
+    void begin() {
+        is_transaction_ = true;
+    }
+
+    void commit() {
+        if (current_offset > offset) {
+            code_generator.sub(code_generator.rsp, current_offset - offset);
+            offset = current_offset;
+        } else if (offset > current_offset) {
+            code_generator.add(code_generator.rsp, offset - current_offset);
+            offset = current_offset;
+        }
+        is_transaction_ = false;
+        for (auto& alloc : allocations) {
+            alloc->is_transaction = false;
+        }
+    }
+
+    x64::jit_generator& code_generator;
+    const Xbyak::Reg base_pointer;
+
+    bool is_transaction_{};
+    size_t offset{};
+    size_t current_offset{};
+    size_t alignment{};
+    std::deque<Allocation::Ptr> allocations{};
```

Comment:
Not really, `std::deque<>` allows to reuse already freed allocation block. It would be much hard to do with `std::stack<>`
How reuse of freed block is done,see:
https://github.com/openvinotoolkit/openvino/pull/13132/files#diff-ad5c4db2706d50420066fbb4beedb2fac0532bf2ba5468e6a9dd53eedbbaa6b7R131

Reply:
Yes, right. We need traversal, in that case we can just use `std::vector`, since we do not need to `push_front()` actually.

Reply:
Changed to `std::vector`